src/main/java/com/thoughtworks/android/AndroidManager.java

```diff
@@ -106,5 +107,28 @@ public Device getDevice(String udid) throws Exception {
         );
     }
 
+    public void startADBLog(String udid, String filePath) throws Exception {
+        cmd.execForProcessToExecute("adb -s " + udid + " logcat -b all -c");
+        process = cmd.execForProcessToExecute("adb -s " + udid + " logcat > " + filePath);
```

Comment:
@shridharkalagi Have you tried this? Isn't it just for clearing all the buffers? Does it retrieve too?

Reply:
@SrinivasanTarget he is clearing the buffer before creating a new one .

Reply:
command to retrieve?

Reply:
adb -s " + udid + " logcat > " + filePath

Reply:
"adb -s udid logcat -b all -c"  will clear all the buffers

Reply:
adb shell logcat -b all -g --to print all the buffers with current buffer size